Automattic/wp-calypso
# composite-checkout: Force features to reload when checkout is complete

#### Changes proposed in this Pull Request

* We're moving Calypso to rely on feature checks more often. Each site's features are loaded from the features endpoint, kicked off by the `<QuerySiteFeatures>` component.
* **Unfortunately, directly after a purchase, Calypso can still have your old set of features in local state**! It feels like you just paid for nothing, until you F5 or switch selected sites.
* In this PR, I made the `checkout-complete-redirecting.tsx` component force features to be reloaded. **However, I'm not really sure about the organization or separation of concerns here**. This gets the job done, but it might not be the best way. Any advice from @sirbrillig is appreciated.
  * Update: I have reworked the approach and I'm feeling much better about it now.

#### Testing instructions

* First, try without this PR applied
* Have a free site
* Visit Settings -> General
* Notice upsell banners on feature credit

![2022-05-16_11-46](https://user-images.githubusercontent.com/937354/168643640-345c4b7b-0f21-4bbd-97cd-a39b4b427706.png)

* Visit Upgrades -> Plans, click "Start with Pro"
* Click the "Pay" button to complete checkout
* You're now on the "Congratulations on your purchase!" page. Click "View my new features"
* Click on Settings -> General
* The upsell banner telling you to buy Pro is still there, even though you have pro

![2022-05-16_11-38](https://user-images.githubusercontent.com/937354/168643852-22ba697a-dc9f-4fd2-ac56-f4dbbc7c9df4.png)

* Apply this PR, then remove the pro purchase and try the steps again. This time, the banner should no longer appear after the purchase.
